client/js/render.js

```diff
@@ -305,18 +315,24 @@ function loadMoreHistory(entries) {
 	});
 }
 
-sidebar.on("click", ".collapse-network", (e) => {
-	const collapseButton = $(e.target);
+sidebar.on("click", "button.collapse-network", (e) => collapseNetwork($(e.target)));
 
+function collapseNetwork(target) {
+	const collapseButton = target.closest(".collapse-network");
+	const networks = JSON.parse(storage.get("thelounge.networks.collapsed")) || {};
+	const networkuuid = collapseButton.closest(".network").data("uuid");
 	collapseButton.closest(".network").toggleClass("collapsed");
 
 	if (collapseButton.attr("aria-expanded") === "true") {
 		collapseButton.attr("aria-expanded", false);
 		collapseButton.attr("aria-label", "Expand");
+		networks[networkuuid] = true;
```

Comment:
What about using a JS `Set` instead?

- In local storage, it would be stored as an array. When loading from the array, use `new Set(array)`. When writing in local storage, use `JSON.stringify([...set])` or `JSON.stringify(Array.from(set))`
- `if (collapsed[key])` becomes `if (collapsed.has(key))`
- `networks[networkuuid] = true` becomes `networks.add(networkuuid)`
- `Object.keys(collapsed).forEach((key)` becomes `collapsed.forEach((key)`

In terms of performance, users with 10k networks might notice a lag of 15ms 😁

Reply:
Sure

Reply:
Euch, I just realised this'll definitely increase complexity. It goes from O(1) to O(n). I know you don't think that's a big issue, but ... just pointing it out.

Reply:
Which part? Isn't JS `Set.has` O(1)?
Also in terms of performance, it's a mixed bag because `collapsed.forEach` is going to go be faster than `Object.keys(collapsed).forEach`, but realistically, in this context, O(n) approximates to O(1) (unlikely someone will have more than... 15 networks, I'd say?)

Reply:
It seems like a lot of unnecessary shuffling to add `Set` here. If these were natively stored in JSON then this would be great, but in this instance just a plain object works fine.

Reply:
Plain array would work fine yeah. `Set` is not that bad though, one serialization + one deserialization 🤷‍♂️
But this would be more justified if duplicates was a concern. In the current implementation, it's not.

Reply:
> Which part? Isn't JS Set.has O(1)?

It depends on the implementation, which I can't be bothered finding, but I assume it has to iterate over all the elements in the set to find it if it's in, which would be O(n).

> Also in terms of performance, it's a mixed bag because collapsed.forEach is going to go be faster than Object.keys(collapsed).forEach, 

True, but this happens once every load, whereas checking .has each time we collapse/expand is a lot more.

> but realistically, in this context, O(n) approximates to O(1) (unlikely someone will have more than... 15 networks, I'd say?)

That's not how it works :-P 

You can say that going to O(n) doesn't impact performance too much, sure, but ... it doesn't "approximate to O(1)" by definition, it is much worse :-P 

I don't really care, as you mentioned, it's not like the performance hit is going to be big, and I don't really care.

The code for Set seems a bit more straightforward.

Reply:
> That's not how it works :-P

Psssh, [you had to be that guy](https://xkcd.com/386/) :)
I stand by it though. You're saying this statement is wrong, and I'm saying notion of complexity and big-O notation, _in this context_ does not even apply, considering the input size.

Anywayyyy, bikeshedding aside, it looks good! I didn't realize that `Set`s were not always O(1) (TIL: spec apparently requires < O(n), so implementations can be either O(1) or O(log(n))!), and in our input size, plain arrays would do just just fine, up to you.